src/emc/usr_intf/stepconf/stepconf.py

```diff
@@ -918,26 +918,72 @@ def find_parport(self):
+        for line in in_file:
+            if "parport" in line:
+                tmprow = line.strip()
+                lrow = tmprow.split(":")
+                address_range = lrow[0].strip()
+                init_address = address_range.split("-")[0].strip()
+                if(init_address != "0000"):
+                    lparport.append("0x" + init_address)
+        in_file.close()
+        """
         try:
             in_file = open("/proc/ioports","r")
+            try:
+                for line in in_file:
+                    if "parport" in line:
+                        tmprow = line.strip()
+                        lrow = tmprow.split(":")
+                        address_range = lrow[0].strip()
+                        init_address = address_range.split("-")[0].strip()
+                        if(init_address != "0000"):
+                            lparport.append("0x" + init_address)
+                in_file.close()
+                # if I have found ports then return
+                if (lparport != []):
+                    return(lparport)
+            except err:
+                print ("Error read parport in /proc/ioports")
+                in_file.close()
         except:
             print ("Unable to open /proc/ioports")
-            return([])
 
-        try:
-            for line in in_file:
-                if "parport" in line:
-                    tmprow = line.strip()
-                    lrow = tmprow.split(":")
-                    address_range = lrow[0].strip()
-                    init_address = address_range.split("-")[0].strip()
-                    lparport.append("0x" + init_address)
-        except:
-            print ("Error find parport")
-            in_file.close()
-            return([])
-        in_file.close()
         if lparport == []:
+            # Try /proc/sys/dev/parport/parport#/base-addr
+            parport_path="/proc/sys/dev/parport"
+            if(os.path.isdir(parport_path)):
+                parport_list=os.listdir(parport_path)
+                for current_parport in parport_list:
+                    if(current_parport == "default"):
+                        continue
+                    # TODO
+                    # find base-addr file
+                    baseaddr=os.path.join(parport_path, current_parport, "base-addr")
+                    if(os.path.exists(baseaddr) == True):
+                        try:
+                            in_file = open(baseaddr,"r")
+                        except:
+                            print ("Unable to open %s" % baseaddr )
+                            continue
+                        # read base-addr file
+                        try:
+                            for line in in_file:
+                                # get init_address and number of port
+                                lline=line.split()
+                                dec_address=lline[0].strip()
+                                port_number=lline[1].strip()
```

Comment:
The code says here *"number of port"* but I couldn't find that on the [documentation](https://www.kernel.org/doc/html/v5.6/admin-guide/parport.html) which mention that this file contains only addresses. In my case the file contains "888 1912" _(kernel : 5.6.17rt9)_.

Apart from that running `stepconf` on the existing configuration changed it as following:
```
$ git diff parport.stepconf parport/parport.hal
diff --git a/parport.stepconf b/parport.stepconf
index 352a87f..5e351d7 100644
--- a/parport.stepconf
+++ b/parport.stepconf
@@ -28,13 +28,14 @@
   <property name="floatsout" type="float" value="10.0"/>
   <property name="halui" type="bool" value="False"/>
   <property name="halui_list" type="eval" value="[]"/>
-  <property name="ioaddr" type="string" value="0x378"/>
+  <property name="ioaddr" type="string" value="0"/>
   <property name="ioaddr2" type="string" value="1"/>
   <property name="ioaddr3" type="string" value="2"/>
   <property name="ladderconnect" type="bool" value="True"/>
   <property name="ladderhaltype" type="int" value="0"/>
   <property name="laddername" type="string" value="custom.clp"/>
   <property name="latency" type="float" value="22000.0"/>
+  <property name="lparport" type="eval" value="['0']"/>
   <property name="machinename" type="string" value="parport"/>
   <property name="manualtoolchange" type="bool" value="True"/>
   <property name="md5sums" type="eval" value="[]"/>
diff --git a/parport/parport.hal b/parport/parport.hal
index ff63696..0cb56a1 100644
--- a/parport/parport.hal
+++ b/parport/parport.hal
@@ -1,9 +1,9 @@
-# Generated by stepconf 1.1 at Sat Jun  6 13:56:45 2020
+# Generated by stepconf 1.1 at Mon Jun 15 22:59:59 2020
 # If you make changes to this file, they will be
 # overwritten when you run stepconf again
 loadrt [KINS]KINEMATICS
 loadrt [EMCMOT]EMCMOT base_period_nsec=[EMCMOT]BASE_PERIOD servo_period_nsec=[EMCMOT]SERVO_PERIOD num_joints=[KINS]JOINTS
-loadrt hal_parport cfg="0x378 out"
+loadrt hal_parport cfg="0 out"
 loadrt stepgen step_type=0,0,0
 loadrt pwmgen output_type=1
 ```

So the `0x378` value is basically gone.

Reply:
888 decimal == 378 hex.
1912 decimal == 778 hex.
Do you have a card with two parallel ports? Can you post how your directory is made?
**ls /proc/sys/dev/parport/**

Nicola.

Reply:
>Do you have a card with two parallel ports?

No, it is built-in parallel port.

>Can you post how your directory is made?

```
$ ls -R /proc/sys/dev/parport/
/proc/sys/dev/parport/:
default  parport0

/proc/sys/dev/parport/default:
spintime  timeslice

/proc/sys/dev/parport/parport0:
autoprobe  autoprobe0  autoprobe1  autoprobe2  autoprobe3  base-addr  devices  dma  irq  modes  spintime

/proc/sys/dev/parport/parport0/devices:
active

```